### PR TITLE
First major round of development: pipeline design + meta/data ingestion + plotting functions + jupyter notebook

### PR DESCRIPTION
**Pull request by ttngu207**
+ The pipeline here mirrors MAP's pipeline for the most part (some minor tweaks)
+ Ingestion of Metadata and ephys data
+ Plotting functions also mostly reuse/rework from MAP
+ Jupyter notebook with figures in the paper reproduced: fig2b, fig2c, supplemental fig2

Next:
+ NWB export
+ Write README and further polish this repository

### CODE REVIEW

File: pipeline/experiment.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    -> Session
+    -> TaskProtocol
+    """
+
+
+@schema
+class SessionComment(dj.Manual):
+    definition = """
+    -> Session
+    session_comment : varchar(767)
```
Comment (shenshan) [inline pipeline/experiment.py]: `session_comment` should probably be a secondary attribute here.

File: pipeline/lab.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+    class Notes(dj.Part):
+        definition = """
+        # Notes for virus
+        -> Virus
+        note_id     : int
+        ---
+        note        : varchar(256)
+        """
+
```
Comment (shenshan) [inline pipeline/lab.py]: Looks like there is another Virus table in the schema `virus`. Delete this one?

Comment (shenshan) [reply pipeline/lab.py]: Also the `SurgeryVirusInjection`

File: notebooks/Untitled.ipynb [hunk 3]
```diff
@@ -0,0 +1,663 @@
+{
```
Comment (eywalker) [inline notebooks/Untitled.ipynb]: Either do not commit this or give it some useful name.

File: pipeline/experiment.py (hunk 1 above)
Comment (eywalker) [reply pipeline/experiment.py]: Indeed, `session_comment` should be a non-pk attribute. Even if multiple comments are allowed per session, another PK attribute like `comment_id: tinyint` ought to be used.

**Review by eywalker — CHANGES_REQUESTED** [review]

File: pipeline/lab.py (hunk 2 above)
Comment (ttngu207) [reply pipeline/lab.py]: Those are from the MAP pipelines, perhaps not applicable here, I have removed

File: pipeline/experiment.py (hunk 1 above)
Comment (ttngu207) [reply pipeline/experiment.py]: fixed, must be a mistake (forgetting the ---), yet this same error is in MAP still

File: notebooks/Untitled.ipynb (hunk 3 above)
Comment (ttngu207) [reply notebooks/Untitled.ipynb]: fixed

**Pull request merged**